renderers/chord_hover.js

```diff
@@ -3,19 +3,103 @@
 const chordDiagram = require('./chord_diagram.js');
 const { guitarChordLibrary } = require('./chord_library.js');
 
+const slowScrollFunction = `
+function slowScroll(id, direction, distance, step) {
+  var element = document.getElementById(id);
+  
+  if (element) {
+    var scrollAmount = 0;
+    var slideTimer = setInterval(function() {
+      element.scrollLeft += direction === 'left' ? -step : step;
+
+      scrollAmount += step;
+      if(scrollAmount >= distance){
+          window.clearInterval(slideTimer);
+      }
+    });
+  }
+}`;
+
+const incrementIdFunction = `
+function incrementId(id, direction) {
+  var countId = id + '-count'
+  var element = document.getElementById(countId);
+
+  if (element) {
+    var values = element.innerHTML.split(' of ');
+    var currentValue = parseInt(values[0]);
+
+    if (direction === 'left' && currentValue > 1) {
+      element.innerHTML = currentValue - 1;
+      element.innerHTML += ' of ' + values[1];
+    } else if (direction === 'right' && currentValue < parseInt(values[1])) {
+      element.innerHTML = currentValue + 1;
+      element.innerHTML += ' of ' + values[1];
+    }
+  }
+}
+`;
+
+function createDiagramDiv(voicing, chordDiagramContainerDiv) {
+  const svg = chordDiagram.renderChordDiagram(voicing);
+
+  const chordDiagramDiv = document.createElement('div');
+
+  chordDiagramDiv.className = 'diagram';
+  chordDiagramDiv.innerHTML += svg;
+
+  chordDiagramContainerDiv.appendChild(chordDiagramDiv);
+}
+
 function addChordToDiv(voiceDiv, chord) {
   voiceDiv.className += ' chord';
 
   if (guitarChordLibrary.has(chord.toString())) {
-    const chordDiagramDiv = document.createElement('div');
-    chordDiagramDiv.className = 'diagram';
+    const containerDiv = document.createElement('div');
+    containerDiv.className = 'diagram-container';
+
+    const contentDiv = document.createElement('div');
+    contentDiv.className = 'diagram-content-container';
+
+    const id = 'chord' + Math.floor(Math.random() * 10000);
+    contentDiv.id = id;
+
+    if (!document.getElementById('chord-hover-script')) {
```

Comment:
Ack! It's a hack!

I'm okay with this implementation, but there's a better way. We essentially want to inject this code as a header to the HTML. I think we can do it by using a renderer:
https://github.com/markdown-it/markdown-it/blob/master/docs/architecture.md#renderer

Let's just add a TODO for now to investigate, but I do want to circle back and use Rulers to inject our javascript, as well as (possibly) do multiple columns. That being said, I like the simplest possible approach, and I'm coming around to your idea of implementing columns as a feature of music-markdown instead of markdown-it-music.

Reply:
Yeah, definitely a hack 😢 I'm also okay with waiting to submit this until I understand the renderer rules because I'd love to have it done a better way XD

From what I understand now, we would create a renderer rule that would inject this script into the HTML once, instead of doing it in part of our render code? I'm not sure how the header part fits in though.

Reply:
No wait! Add TODO. Fix later.